src/include/storage/index/index.h

```diff
@@ -116,7 +108,9 @@ class Index {
    * @param[out] value_list the values associated with the keys
    */
   virtual void ScanAscending(const transaction::TransactionContext &txn, const ProjectedRow &low_key,
-                             const ProjectedRow &high_key, std::vector<TupleSlot> *value_list) = 0;
+                             const ProjectedRow &high_key, std::vector<TupleSlot> *value_list) {
```

Comment:
Same here.

Reply:
Similar to GC: not all indexes will support this operation. Pure virtual forces the derived types to implement it when they might not need to.